kiwi/xml_state.py

```diff
@@ -503,10 +503,8 @@ def get_bootstrap_archives(self) -> List:
         bootstrap_packages_sections = self.get_bootstrap_packages_sections()
         if bootstrap_packages_sections:
             for bootstrap_packages_section in bootstrap_packages_sections:
-                archive_list = bootstrap_packages_section.get_archive()
-                if archive_list:
-                    for archive in archive_list:
-                        result.append(archive.get_name().strip())
+                for archive in bootstrap_packages_section.get_archive():
+                    result.append(archive.get_name().strip())
```

Comment:
I think this change is unrelated to the support for target_dir in ```<archive>``` With the version of generateDS as you used it it's correct that section lists with no value return an empty list [] rather than None. However it's not guaranteed that an older version of generateDS is used to create the XML API and then this change can lead to unexpected failures. This is because if you do this:

```python
x = None

for f in x:
    print(f)
```

you get an error in python. I coded it to be safe in any direction even though that requires one more line of code :) If you don't mind can we drop this change as it's unrelated ?

Thanks